Bottom: 91153ae24e0243a1f3cf380e415d9bc5e8094789
Top:    00e1df4bcbc33b38df0e85b87fbbabd2da64d772
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-09 14:58:14 -0700

rsocket: Build rspreload library as part of build

Build rspreload.so, but do not install it.  This avoids the
need for the user to build it manually, without risking that
the preload library intercepts socket calls without the user's
explicit knowledge.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/Makefile.am b/Makefile.am
index 51b2f89..901e8ee 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,11 +1,12 @@
 INCLUDES = -I$(srcdir)/include
 
-lib_LTLIBRARIES = src/librdmacm.la
+lib_LTLIBRARIES = src/librdmacm.la src/librspreload.la
 
 ACLOCAL_AMFLAGS = -I config
 AM_CFLAGS = -g -Wall -D_GNU_SOURCE -DSYSCONFDIR=\"$(sysconfdir)\" -DRDMADIR=\"@rdmadir@\"
 
 src_librdmacm_la_CFLAGS = $(AM_CFLAGS)
+src_librspreload_la_CFLAGS = $(AM_CFLAGS)
 
 if HAVE_LD_VERSION_SCRIPT
     librdmacm_version_script = -Wl,--version-script=$(srcdir)/src/librdmacm.map
@@ -19,6 +20,9 @@ src_librdmacm_la_LDFLAGS = -version-info 1 -export-dynamic \
 			   $(librdmacm_version_script)
 src_librdmacm_la_DEPENDENCIES =  $(srcdir)/src/librdmacm.map
 
+src_librspreload_la_SOURCES = src/preload.c src/indexer.c
+src_librspreload_la_LIBADD = $(top_builddir)/src/librdmacm.la
+
 bin_PROGRAMS = examples/ucmatose examples/rping examples/udaddy examples/mckey \
 	       examples/rdma_client examples/rdma_server examples/rdma_xclient \
 	       examples/rdma_xserver examples/rstream examples/rcopy
